packages/labextension/src/widget.tsx

```diff
@@ -0,0 +1,268 @@
+          setKernelId(newKernelId);
+        }
+      };
+    };
+
+    const notebookChangeCallback = (
+      sender: INotebookTracker,
+      panel: NotebookPanel | null
+    ) => {
+      if (panel === null) {
+        // Ideally we would switch to a new "select a notebook to get kernel
+        // usage" screen instead of showing outdated info.
+        return;
+      }
+      if (kernelChangeCallback) {
+        kernelChangeCallback.panel.sessionContext.kernelChanged.disconnect(
+          kernelChangeCallback.callback
+        );
+      }
+      kernelChangeCallback = {
+        callback: createKernelChangeCallback(panel),
+        panel,
+      };
+      panel.sessionContext.kernelChanged.connect(kernelChangeCallback.callback);
+
+      if (panel.sessionContext.session?.kernel?.id !== kernelId) {
+        const kernelId = panel.sessionContext.session?.kernel?.id;
+        if (kernelId) {
+          setKernelId(kernelId);
+          const path = panel.sessionContext.session?.model.path;
+          setPath(path);
+          requestUsage(kernelId).then((usage) => setUsage(usage));
+        }
+      }
+    };
+    props.currentNotebookChanged.connect(notebookChangeCallback);
+    return () => {
+      props.currentNotebookChanged.disconnect(notebookChangeCallback);
+      // In the ideal world we would disconnect kernelChangeCallback from
+      // last panel here, but this can lead to a race condition. Instead,
+      // we make sure there is ever only one callback active by holding
+      // it in a global state.
+    };
+  }, [kernelId]);
+
+  if (kernelId) {
+    if (usage) {
+      return !usage.hostname ? (
+        <>
+          <h3 className="jp-KernelUsage-section-separator">
+            {props.trans.__('Kernel usage details are not available')}
+          </h3>
+          <div className="jp-KernelUsage-section-separator">
+            {props.trans.__(
+              'Please check with your system administrator that you running IPyKernel version 6.10.0 or above.'
+            )}
+          </div>
+        </>
+      ) : (
+        <>
+          <h3 className="jp-KernelUsage-section-separator">
+            {props.trans.__('Kernel usage')}
+          </h3>
+          <div className="jp-KernelUsage-separator">
+            {props.trans.__('Kernel Host')}: {usage.hostname}
+          </div>
+          <div className="jp-KernelUsage-separator">
+            {props.trans.__('Notebook')}: {path}
+          </div>
+          <div className="jp-KernelUsage-separator">
+            {props.trans.__('Kernel ID')}: {kernelId}
+          </div>
+          <div className="jp-KernelUsage-separator">
+            {props.trans.__('Timestamp')}: {usage.timestamp?.toLocaleString()}
+          </div>
+          <div className="jp-KernelUsage-separator">
+            {props.trans.__('Process ID')}: {usage.pid}
+          </div>
+          <div className="jp-KernelUsage-separator">
+            {props.trans.__('CPU')}: {usage.kernel_cpu}% used
```

Comment:
`% used` not translated.

```suggestion
            {props.trans.__('CPU')}: {props.trans.__('%s %% used', usage.kernel_cpu)}
```

Reply:
You may want to be even more aggressive here and include `: ` part. Gettext docs highlights that:

>  But ‘"%s: %d"’ does require translation, because in French, unlike in English, it’s customary to put a space before a colon.

I have no knowledge of French but sounds like something users may be interested in.